src/els_parser.erl

```diff
@@ -25,46 +25,47 @@ parse(Text) ->
 
 -spec parse_file(file:io_device()) -> {ok, [poi()]}.
 parse_file(IoDevice) ->
-  {ok, Forms} = els_dodger:parse(IoDevice, {1, 1}),
-  Tree = erl_syntax:form_list(Forms),
-  %% Reset file pointer position.
-  {ok, 0} = file:position(IoDevice, 0),
-  {ok, Extra} = parse_attribute_pois(IoDevice, [], {1, 1}),
+  {ok, NestedPOIs} = els_dodger:parse(IoDevice, {1, 1}, fun parse_form/3, []),
   ok = file:close(IoDevice),
-  POIs = points_of_interest(Tree),
-  {ok, Extra ++ POIs}.
+  {ok, lists:flatten(NestedPOIs)}.
 
-%%==============================================================================
-%% Internal Functions
-%%==============================================================================
+-spec parse_form(file:io_device(), any(), [any()]) ->
+    {'ok', erl_syntax:forms()
+  | none, integer()}
+  | {'eof', integer()}
+  | {'error', any(), integer()}.
+parse_form(Dev, L0, Options) ->
```

Comment:
This seems a re-implementation of `els_dodger:parse_form/3`. Why not to expose that one, instead?

Reply:
Sorry, I was misreading the code. Ignore this comment.

Reply:
An alternative option would have been to add a callback function to the `Options` list which could have been executed from within the `els_dodger`, and have the `parse_form/4` return the final _accumulator_ together with the list of forms.

Reply:
That would have required a change in the implementation of dodger which I was trying to avoid.